dll/directx/wine/dpnhpast/dpnhpast.rgs

```diff
@@ -0,0 +1,15 @@
+HKCR
+{
+    NoRemove Interface
+    {
+    }
+    NoRemove CLSID
+    {
+        '{963AB779-16A1-477C-A36D-CB5E711938F7}' = s 'DirectPlay NAT Helper PAST object'
+        {
+            InprocServer32 = s '%MODULE%' { val ThreadingModel = s 'Both' }
```

Comment:
Why Both? (Where did you get this from)

Reply:
I made this file manually, same as it is present in other Wine DirectX dlls in the ROS source code. It's just registry keys and values which are associated with dpnhpast.dll. I thought at first that it's required file since *.rgs is ReactOS-specific format (absent in Wine sources). But as I see now, such values are present only on clear XP installation, but not on 2003. And they are also absent in Wine/ReactOS. So I probably need to remove this file.
![dpnhpast_w2k3_registry](https://user-images.githubusercontent.com/26385117/55949013-5c7cb700-5c5a-11e9-9433-a70d83f32c3a.png)

Reply:
If you completely remove this .rgs file, you can also remove the entry in media/inf/syssetup.inf

Reply:
For future reference: these files can be extracted from the .res files that Wine generates.
However they only make sense in modules where `DllRegisterServer` calls `__wine_register_resources`, otherwise it would have no effect.

Reply:
@ThFabba but where Wine generates them? In which directory of wineprefix they are stored?

Reply:
"since *.rgs is ReactOS-specific format " <-- it's a resource embedded in the DLL, and this is a format that also is on Windows as it's parsed by the Active-Template-Library runtime when the DLL is called for COM registration.

Reply:
OK, I found out that *.res files are generated during Wine compiling, in the build directory/dlls/dllname. But there are only version.res files which are compiled from version.rc, and contain only VersionInfo. Where I can get other *.res?... I hope to get exact code for the dpnhpast.rgs. Current rgs is probably incorrect.
![dpnhpast_res](https://user-images.githubusercontent.com/26385117/55988987-e86bfe80-5cad-11e9-8792-ace32f62fc1d.png)

Reply:
Yes, this modules doesn't have classes to register. The way you recognize this in Wine modules is, as I mentioned, a call to `__wine_register_resources`. Another requirement is that `Makefile.in` for that module includes `IDL_SRCS` (though not all modules with idl files also need an rgs)

Reply:
Thanks! I was able to check it out. For the test, I re registered dpnhpast.dll in Wine with enabled `WINEDEBUG=+all` into file, according to https://wiki.winehq.org/Debug_Channels and captured a debug log. +all includes everything, including +relay, which shows API calls. But I didn't find any lines with `__wine_register_resources` in that log. Moreover, there is no `IDL_SRCS` in `Makefile.in` and even any *.idl in Wine's dpnhpast sources folder, let alone rgs. Also that function isn't called in none of the functions in `main.c`. So looks like this `dpnhpast.rgs` is not needed and indeed can be safely removed. I think this test is correct.

Reply:
Oh, I understood! In `main.c` `DllRegisterServer` doesn't call `__wine_register_resources`. Now I can be completely sure that it's right.

Reply:
Removed.